packages/terra-form/src/NumberField.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Field from './Field';
+import Input from './Input';
+
+const propTypes = {
+  /**
+   * The defaultValue of the input element. Use this to create an uncontrolled input
+   */
+  defaultValue: PropTypes.number,
+  /**
+   * Error message for when the input is invalid
+   */
+  error: PropTypes.node,
+  /**
+   *  Help element to display with the input
+   */
+  help: PropTypes.node,
+  /**
+   * Additional attributes to attach to the number input
+   */
+  inputAttrs: PropTypes.object,
+  /**
+   * Whether the field is inline
+   */
+  isInline: PropTypes.bool,
+  /**
+   * Label of the input
+   */
+  label: PropTypes.node,
+  /**
+   * Maximum value allowed for the input
+   */
+  max: PropTypes.number,
+  /**
+   * Minimum value allowed for the input
+   */
+  min: PropTypes.number,
+  /**
+   * Name of the input attribute
+   */
+  name: PropTypes.string,
+  /**
+   * Function to trigger when user changes the input value. Provide a function to create a controlled input.
+   */
+  onChange: PropTypes.func,
+  /**
+   * Whether the input is required
+   */
+  required: PropTypes.bool,
+  /**
+   * Incremental steps for increasing and descreasing the value
+   */
+  step: PropTypes.number,
+  /**
+   * The defaultValue of the input element. Use this to create an uncontrolled input.
+   */
+  value: PropTypes.oneOfType([
```

Comment:
Given that some backend servers may potentially return numbers as strings, I think keeping it as both strings and numbers would make it easier to work with. I tested with a string that is a number, and the input initialized correctly as a number. 

What do others think though?